omnigibson/sensors/vision_sensor.py

```diff
@@ -215,38 +205,57 @@ def initialize_sensors(self, names):
                 If they are not part of self._RAW_SENSOR_TYPES' keys, we will simply pass over them
         """
-        # Standardize the input and grab the intersection with all possible raw sensors
-        names = set([names]) if isinstance(names, str) else set(names)
-        names = names.intersection(set(self._RAW_SENSOR_TYPES.keys()))
-
-        # Initialize sensors
-        sensors = []
+        names = {names} if isinstance(names, str) else set(names)
         for name in names:
-            sensors.append(lazy.omni.syntheticdata.sensors.create_or_retrieve_sensor(self._viewport.viewport_api, self._RAW_SENSOR_TYPES[name]))
-
-        # Suppress syntheticdata warning here because we know the first render is invalid
-        with suppress_omni_log(channels=["omni.syntheticdata.plugin"]):
-            render()
-        render()    # Extra frame required to prevent access violation error
+            self._add_modality_to_backend(modality=name)
 
     def _get_obs(self):
         # Make sure we're initialized
         assert self.initialized, "Cannot grab vision observations without first initializing this VisionSensor!"
 
         # Run super first to grab any upstream obs
-        obs = super()._get_obs()
-
-        # Process each sensor modality individually
-        for modality in self.modalities:
-            mod_kwargs = dict()
-            mod_kwargs["viewport"] = self._viewport.viewport_api
-            if modality == "seg_instance":
-                mod_kwargs.update({"parsed": True, "return_mapping": False})
-            elif modality == "bbox_3d":
-                mod_kwargs.update({"parsed": True, "return_corners": True})
-            obs[modality] = self._SENSOR_HELPERS[modality](**mod_kwargs)
+        obs, info = super()._get_obs()
+
+        for modality in self._modalities:
+            raw_obs = self._annotators[modality].get_data()
+            # Obs is either a dictionary of {"data":, ..., "info": ...} or a direct array
+            obs[modality] = raw_obs["data"] if isinstance(raw_obs, dict) else raw_obs
+            if modality == "seg_semantic":
+                id_to_labels = raw_obs['info']['idToLabels']
+                obs[modality], corrected_id_to_labels = self._remap_semantic_segmentation(obs[modality], id_to_labels)
+                info[modality] = corrected_id_to_labels
+            elif modality == "seg_instance":
+                id_to_labels = raw_obs['info']['idToLabels']
+                info[modality] = id_to_labels
+        return obs, info
+    
+    def _remap_semantic_segmentation(self, img, id_to_labels):
+        """
+        Remap the semantic segmentation image to the class IDs defined in CLASS_NAME_TO_CLASS_ID.
+        Also, correct the id_to_labels input with the labels from CLASS_NAME_TO_CLASS_ID and return it.
+        """
+        # Convert string IDs to integers and find the max ID for array size
+        max_id = max([int(id) for id in id_to_labels.keys()])
```

Comment:
might be more efficient to operate directly on the generator:

`max(id_to_labels.keys())`